src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/hooks.tsx

```diff
@@ -0,0 +1,42 @@
+import { useFormContext } from 'react-hook-form';
+import { useFlatFormErrors } from '~hooks';
+import { ACTION, Action } from '~constants/actions';
+import { REPUTATION_VALIDATION_FIELD_NAME } from '../../hooks/useReputationValidation';
+import { PERMISSIONS_VALIDATION_FIELD_NAME } from '../../hooks/usePermissionsValidation';
+
+export const useGetActionErrors = () => {
+  const { formState } = useFormContext();
+  const hasErrors = !formState.isValid && formState.isSubmitted;
+  const flatFormErrors = useFlatFormErrors(formState.errors).filter(
+    ({ key }) =>
+      ![
+        'title',
+        REPUTATION_VALIDATION_FIELD_NAME,
+        PERMISSIONS_VALIDATION_FIELD_NAME,
+      ].includes(String(key)),
+  );
+
+  return {
+    hasErrors,
+    flatFormErrors,
+  };
+};
+
+export const useGetAdditionalValidationErrors = (
+  action: Action | undefined,
+) => {
+  const {
+    formState: { isDirty },
+  } = useFormContext();
+
+  if (!action) {
+    return undefined;
+  }
+
+  switch (action) {
+    case ACTION.EDIT_COLONY_DETAILS:
+      return isDirty ? undefined : 'actionSidebar.editColonyDetails.error';
+    default:
+      return undefined;
+  }
+};
```

Comment:
I don't like the idea of this being a hook, checking for whether the values have changed can be easily done via validation schema and custom yup tests.